workflows/tests/test_cluster_jobs_integration.py

```diff
@@ -102,39 +131,64 @@ async def test_run_cluster_job_state_persistence(
         file.write("my,initial,params")
 
     # cluster jobs can accept a list of input files to hash
-    job_id_initial_with_hash = await run_cluster_job(
+    job_initial_with_hash = await run_cluster_job(
         name="test job",
         command="echo 'test'",
         expected_time=timedelta(minutes=1),
         memory="100M",
         environment={},
+        resubmit_policy=ResubmitIfFailedPolicy,
+        working_dir=tmp_path,
         input_files_to_hash=[tmp_path / "my_inputs.csv"],
     )
 
     # if input file unchanged, should be same job
-    job_id_repeat_with_hash = await run_cluster_job(
+    job_repeat_with_hash = await run_cluster_job(
         name="test job",
         command="echo 'test'",
         expected_time=timedelta(minutes=1),
         memory="100M",
         environment={},
+        resubmit_policy=ResubmitIfFailedPolicy,
+        working_dir=tmp_path,
         input_files_to_hash=[tmp_path / "my_inputs.csv"],
     )
-    assert job_id_initial_with_hash == job_id_repeat_with_hash
+    assert job_initial_with_hash.cluster_job_id == job_repeat_with_hash.cluster_job_id
 
     # if input file changes, should be new job
     with open(f"{tmp_path}/my_inputs.csv", "w") as file:
         file.write("my,altered,params")
 
-    job_id_repeat_with_hash = await run_cluster_job(
+    job_repeat_with_hash = await run_cluster_job(
         name="test job",
         command="echo 'test'",
         expected_time=timedelta(minutes=1),
         memory="100M",
         environment={},
+        resubmit_policy=ResubmitIfFailedPolicy,
+        working_dir=tmp_path,
         input_files_to_hash=[tmp_path / "my_inputs.csv"],
     )
-    assert job_id_initial_with_hash != job_id_repeat_with_hash
+    assert job_initial_with_hash.cluster_job_id != job_repeat_with_hash.cluster_job_id
+
+    # if input file changes, but policy ignores that, should NOT be new job
```

Comment:
I'm struggling to think of a case when a job that has a modified input should not be triggered

Reply:
I think it would be very niche, yes, and only really added it because I knew it'd be easier to do so with the architecture fresh in mind. Maybe premature over-engineering. A use case would be if we know some input file is non-deterministic, but we don't really mind:

```python
@task(cached=False)
def get_things_from_very_fast_api(study='ERP1') -> Path:
    csv = internet.get(f"my-api.org/{study}).as_csv(f"{study}_data.csv")
    return csv.path

csv_path = get_things_from_very_fast_api(study)

do_expensive_stuff_with_data(
    study,
    csv_path,
    resubmit_policy=DontResubmitIfOnlyInputFilesChangePolicy
)  # <-- if this had previously succeeded, we may not care that the csv is now in a different order

```

Reply:
perfect.. I was thinking something like that but couldn't come up with a "real" example of stuff we usually run. All good then, carry on :)